src/test/java/com/ipiecoles/java/java350/service/EmployeServiceTest.java

```diff
@@ -141,4 +143,131 @@ public void testEmbaucheEmployeManagerMiTempsMaster99999(){
         Assertions.assertEquals("Limite des 100000 matricules atteinte !", e.getMessage());
     }
+
+    //TESTER METHODE calculPerformanceCommercial
+    //Tester si matricule null
+    //Tester si cA traite  null
+    //Tester si objectif cA null
+    //Tester si cA traite est négatif
+    //Tester si objectif cA négatif
+    @Test
+    public void testCalculPerformanceCommercialMatriculeIsNull(){
+        //Given
+        String matricule = null;
+        Long caTraite = 15L;
+        Long objectifCa = 15L;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("Le matricule ne peut être null et doit commencer par un C !", e.getMessage());
+    }
+
+    @Test
+    public void testCalculPerformanceCommercialMatriculeNotStartWithC(){
+        //Given
+        String matricule = "M00001";
+        Long caTraite = 15L;
+        Long objectifCa = 15L;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("Le matricule ne peut être null et doit commencer par un C !", e.getMessage());
+    }
+    @Test
+    public void testCalculPerformanceCommercialcATraiteIsNull(){
+        //Given
+        String matricule = "C00001";
+        Long caTraite = null;
+        Long objectifCa = 15L;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("Le chiffre d'affaire traité ne peut être négatif ou null !", e.getMessage());
+    }
+    @Test
+    public void testCalculPerformanceCommercialObjectifCaIsNull(){
+        //Given
+        String matricule = "C00001";
+        Long caTraite = 15L;
+        Long objectifCa = null;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("L'objectif de chiffre d'affaire ne peut être négatif ou null !", e.getMessage());
+    }
+
+    @Test
+    public void testCalculPerformanceCommercialCaTraiteIsNegative(){
+        //Given
+        String matricule = "C00001";
+        Long caTraite = -15L;
+        Long objectifCa = 15L;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("Le chiffre d'affaire traité ne peut être négatif ou null !", e.getMessage());
```

Comment:
Préférer **AssertJ** par rapport à **Junit**

```suggestion
        Assertions.assertThat(e.getMessage()).isEqualTo("Le chiffre d'affaire traité ne peut être négatif ou null !");
```